rllib/env/multi_agent_env.py

```diff
@@ -132,7 +135,15 @@ def observation_space_contains(self, x: MultiAgentDict) -> bool:
                 self._check_if_space_maps_agent_id_to_sub_space()
             )
         if self._spaces_in_preferred_format:
-            return self.observation_space.contains(x)
+            if allow_partial_multi_agent_obs:
+                # If we allow partial observations, only check whether observations of
+                # agents present in x are contained
+                return any(
```

Comment:
wait this fixes edward's bug, but is not the correct way to go about this. 

IIRC his bug is that he has observations that only come from some of his agents, but not all of them, so when he goes to do space checking, the check fails because he doesn't have all the agent keys. 


what we should do instead is this:

```python
if self._spaces_in_preferred_format:
    for key, agent_obs in x.items():
        if not self.observation_space[key].contains(agent_obs):
            return False
    return True
```
and then we also don't need `allow_partial_multi_agent_obs`